### PR TITLE
ui: self curation - editing drawer

### PR DESCRIPTION
**Pull request by karolina-siemieniuk-morawska**
* ref: cern-sis/issues-inspire #237

ezgif com-video-to-gif

### CODE REVIEW

File: ui/src/common/components/RouteOrRedirect.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ function RouteOrRedirect({
     <Route
       {...rest}
       render={props =>
-        condition ? <Component {...props} /> : <Redirect to={redirectTo} />
+        condition ? <Component {...props} /> : <Redirect to={redirectTo || '/'} />
```
Comment (drjova) [inline ui/src/common/components/RouteOrRedirect.tsx:20]: 🤔 why do we need that?

**Comment by drjova** [comment]
thank you @karolina-siemieniuk-morawska! @michamos is the wording in the drawer ok for you?

**Comment by drjova** [comment]
@karolina-siemieniuk-morawska we on "Connect" click we should close the drawer, similar to what we do on conference assign :)

File: ui/src/common/components/RouteOrRedirect.tsx (hunk 1 above)
Comment (karolina-siemieniuk-morawska) [reply ui/src/common/components/RouteOrRedirect.tsx:20]: So recently I fixed almost all console errors from external libraries, but missed this one (property to is required in <Redirect /> component from connected-react-router but we sometimes pass undefined which creates error in console). I encountered it while developing so I added this simple fix additionally.

File: ui/src/common/components/LinkLikeButton/LinkLikeButton.tsx [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+  color: string;
+  disabled: boolean;
 }) => (
-  <Button type="text" data-test-id={dataTestId} onClick={onClick} className={`__LinkLikeButton__ ${color}`} data-testid={dataTestId}>
+  <Button
+    disabled={disabled}
+    type="text"
+    data-test-id={dataTestId}
+    onClick={onClick}
+    className={`__LinkLikeButton__ ${color} ${disabled ? 'disabled' : ''}`}
```
Comment (drjova) [inline ui/src/common/components/LinkLikeButton/LinkLikeButton.tsx:24]: `classnames` library :)

**Comment by michamos** [comment]
> thank you @karolina-siemieniuk-morawska! @michamos is the wording in the drawer and button ok for you?

I would change the wording in the drawer to `Find the correct reference:`

File: ui/src/common/components/LinkLikeButton/LinkLikeButton.less [hunk 3]
```diff
@@ -25,4 +25,7 @@
   &.big {
     font-size: 1.5rem;
   }
+  &.disabled {
+    color: #00000050 !important;
```
[the remark this answers is not in the document]
Comment (karolina-siemieniuk-morawska) [reply ui/src/common/components/LinkLikeButton/LinkLikeButton.less:29]: yes, it's hex with opacity

File: ui/src/common/components/LinkLikeButton/LinkLikeButton.tsx [hunk 4]
```diff
@@ -1,27 +1,38 @@
...
+  color: string;
+  disabled: boolean;
 }) => (
-  <Button type="text" data-test-id={dataTestId} onClick={onClick} className={`__LinkLikeButton__ ${color}`} data-testid={dataTestId}>
+  <Button
+    disabled={disabled}
+    type="text"
+    data-test-id={dataTestId}
+    onClick={onClick}
+    className={classNames('__LinkLikeButton__', color, disabled ? 'disabled' : '')}
```
[the remark this answers is not in the document]
Comment (drjova) [reply ui/src/common/components/LinkLikeButton/LinkLikeButton.tsx:25]: https://github.com/JedWatson/classnames#usage :)

**Review by drjova — APPROVED** [review]

**Pull request merged**